src/components/container-query/container-query.tsx

```diff
@@ -0,0 +1,22 @@
+import { useRef } from 'react'
+
+let baseId = 1
+
+export function ContainerQuery({ children, conditions, not = false, containerName = 'container' }) {
```

Comment:
> todo: Please define a prop interface for this component 🙏 Also, can we call this component CSSContainerQuery, just to help make it clear to readers that this component is related to CSS container queries, not a React Query kind of component.

yes, sure already on my todo but I forgot to add it 😅

> todo: Looking at the [MDN @container docs](https://developer.mozilla.org/en-US/docs/Web/CSS/@container#values) more closely, the syntax for the @container rule is @container <container-condition> { ... } where <container-condition> consists of an optional <container-name> and a <container-query>. As such, I think we should make the following changes:

since I separate this component outside the `topBar` so it can be used anywhere I just thought that by providing `containerName` it would be useful for complex usage, e.g nested container 🤷‍♂️ 

> Rename conditions to condition;
Remove containerName and not. These can just be specified as part of the condition prop;
When demonstrating the use of this in TopBar, don't specify a container name (it works fine in my CodeSandbox without specifying it). This just helps us avoid the awkwardness of TopBar consumers wanting to know the container name use by the Main Nav component.

yes, sure 👍